src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -2578,670 +2620,1067 @@ void gfx_set_framebuffer(int fb, float noise_scale);
-                    exec_stack.closeDisp();
-                }
-            }
-        } break;
-        case G_MTX: {
-            uintptr_t mtxAddr = cmd->words.w1;
-
-            if (mtxAddr == SEG_ADDR(0, 0x12DB20) || // GC MQ D
-                mtxAddr == SEG_ADDR(0, 0x12DB40) || // GC NMQ D
-                mtxAddr == SEG_ADDR(0, 0xFBC20) ||  // GC PAL
-                mtxAddr == SEG_ADDR(0, 0xFBC01) ||  // GC MQ PAL
-                mtxAddr == SEG_ADDR(0, 0xFCD00) ||  // PAL1.0
-                mtxAddr == SEG_ADDR(0, 0xFCD40)     // PAL1.1
-            ) {
-                mtxAddr = clearMtx;
-            }
+    if (texAddr == 0) {
+        gfx_texture_cache_clear();
+    } else {
+        gfx_texture_cache_delete((const uint8_t*)texAddr);
+    }
+    return false;
+}
 
-#ifdef F3DEX_GBI_2
-            gfx_sp_matrix(C0(0, 8) ^ G_MTX_PUSH, (const int32_t*)seg_addr(mtxAddr));
-#else
-            gfx_sp_matrix(C0(16, 8), (const int32_t*)seg_addr(cmd->words.w1));
-#endif
-            break;
+bool gfx_noop_handler_f3dex2(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+    const char* filename = (const char*)(cmd)->words.w1;
+    uint32_t p = C0(16, 8);
+    uint32_t l = C0(0, 16);
+    if (p == 7) {
+        g_exec_stack.openDisp(filename, l);
+    } else if (p == 8) {
+        if (g_exec_stack.disp_stack.size() == 0) {
+            SPDLOG_WARN("CLOSE_DISPS without matching open {}:{}", p, l);
+        } else {
+            g_exec_stack.closeDisp();
         }
-        case G_MTX_OTR: {
-            cmd++;
+    }
+    return false;
+}
 
-            uint64_t hash = ((uint64_t)cmd->words.w0 << 32) + cmd->words.w1;
+bool gfx_mtx_handler_f3dex2(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+    uintptr_t mtxAddr = cmd->words.w1;
 
-            int32_t* mtx = (int32_t*)ResourceGetDataByCrc(hash);
+    if (mtxAddr == SEG_ADDR(0, 0x12DB20) || // GC MQ D
+        mtxAddr == SEG_ADDR(0, 0x12DB40) || // GC NMQ D
+        mtxAddr == SEG_ADDR(0, 0xFBC20) ||  // GC PAL
+        mtxAddr == SEG_ADDR(0, 0xFBC01) ||  // GC MQ PAL
+        mtxAddr == SEG_ADDR(0, 0xFCD00) ||  // PAL1.0
+        mtxAddr == SEG_ADDR(0, 0xFCD40)     // PAL1.1
+    ) {
+        mtxAddr = clearMtx;
+    }
+
+    gfx_sp_matrix(C0(0, 8) ^ G_MTX_PUSH, (const int32_t*)seg_addr(mtxAddr));
+
+    return false;
+}
+// Seems to be the same for all other non F3DEX2 microcodes...
+bool gfx_mtx_handler_f3d(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+    uintptr_t mtxAddr = cmd->words.w1;
+
+    if (mtxAddr == SEG_ADDR(0, 0x12DB20) || // GC MQ D
+        mtxAddr == SEG_ADDR(0, 0x12DB40) || // GC NMQ D
+        mtxAddr == SEG_ADDR(0, 0xFBC20) ||  // GC PAL
+        mtxAddr == SEG_ADDR(0, 0xFBC01) ||  // GC MQ PAL
+        mtxAddr == SEG_ADDR(0, 0xFCD00) ||  // PAL1.0
+        mtxAddr == SEG_ADDR(0, 0xFCD40)     // PAL1.1
+    ) {
+        mtxAddr = clearMtx;
+    }
+
+    gfx_sp_matrix(C0(16, 8), (const int32_t*)seg_addr(cmd->words.w1));
+    return false;
+}
+
+// TODO handle special OTR opcodes later...
```

Comment:
what are the "special" opcodes? were they handled before this refactor? maybe list them out or something?